packages/tailwindcss/src/ast.ts

```diff
@@ -339,6 +341,10 @@ export function optimizeAst(ast: AstNode[], designSystem: DesignSystem) {
 
     // AtRule
     else if (node.kind === 'at-rule') {
+      if (node.name === '@keyframes') {
+        context = { ...context, keyframes: true }
```

Comment:
@thecrypticace @RobinMalfait I was wondering if we can clean this up a bit by starting to use an AST walk instead and relying on the parentNode being set correctly to see wether we're inside a keyframe to avoid touching any properties. WDYT? 

The code to remove nodes inside a `context.reference` boundary would become more annoying since we'll have to remove every sibling node instead of removing the whole context node (as context nodes are invisible for traversal).

Reply:
That's an option, but by the time we see a declaration, the parent node will never be the `@keyframes` node, but will be the intermediate rule:
```
@kefyrames {
  0% { /* <- this will be the parent */
    --foo: red;
  }
}
```

Or do you mean that at the the time we reach `@keyframes`, we do a walk in this block, and then mark all found declarations as "don't touch me"?

Reply:
Yeah I was thinking of doing a `return WalkAction.Skip` maybe.